packages/persist/lib/routes/environment/environmentId/postLog.ts

```diff
@@ -0,0 +1,80 @@
+import { z } from 'zod';
+import type { ApiError, Endpoint } from '@nangohq/types';
+import type { EndpointRequest, EndpointResponse, RouteHandler, Route } from '@nangohq/utils';
+import { validateRequest } from '@nangohq/utils';
+import { logContextGetter, oldLevelToNewLevel } from '@nangohq/logs';
+
+type LegacyLogLevel = 'info' | 'debug' | 'error' | 'warn' | 'http' | 'verbose' | 'silly';
+
+const MAX_LOG_CHAR = 10000;
+
+type PostLog = Endpoint<{
+    Method: typeof method;
+    Path: typeof path;
+    Params: {
+        environmentId: number;
+    };
+    Body: {
+        activityLogId: string;
+        level: LegacyLogLevel;
+        msg: string;
+        timestamp?: number | undefined;
+    };
+    Error: ApiError<'post_log_failed'>;
+    Success: never;
+}>;
+
+export const path = '/environment/:environmentId/log';
+const method = 'POST';
+
+const validate = validateRequest<PostLog>({
+    parseBody: (data) =>
+        z
+            .object({
+                activityLogId: z.string(),
+                level: z.enum(['info', 'debug', 'error', 'warn', 'http', 'verbose', 'silly']),
+                msg: z.string(),
+                timestamp: z.number().optional()
+            })
+            .strict()
+            .parse(data),
+    parseParams: (data) =>
+        z
+            .object({
+                environmentId: z.string().transform(Number).pipe(z.number().int().positive()) as unknown as z.ZodNumber
```

Comment:
this `envid` is not parsed the same as in `validate.js` that would be great to unify this